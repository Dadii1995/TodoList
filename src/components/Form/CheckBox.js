import React from 'react'
import { FormGroup, Label, Input } from 'reactstrap'

const CheckBox = ({ field: { name, ...field }, form: { touched, errors }, ...props }) => (
  <FormGroup check>
    <Label check>
      <Input data-cy={props.dataCy} type="checkbox" {...field} name={name} />{' '}
      {props.label && props.label}
    </Label>
  </FormGroup>
)

export default CheckBox
