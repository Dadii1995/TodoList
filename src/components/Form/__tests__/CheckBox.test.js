import renderer from 'react-test-renderer'
import React from 'react'
import { Field, Formik } from 'formik'
import CheckBox from '../CheckBox'

const getWrapper = initialState => {
  const wrapper = renderer.create(
    <Formik initialValues={initialState.initialValues}>
      <Field component={CheckBox} label={initialState.label} name={initialState.name} />
    </Formik>,
  )
  const root = wrapper.root
  return { root, wrapper }
}
describe('<Checkbox/>', () => {
  test('render checkbox with label', () => {
    const initialState = {
      name: 'testCheckbox',
      label: 'Accept terms',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render checkbox without label', () => {
    const initialState = {
      name: 'testCheckbox',
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
  test('render checkbox with initial value', () => {
    const initialState = {
      name: 'testCheckbox',
      initialValues: {
        testCheckbox: true,
      },
    }
    const { wrapper } = getWrapper(initialState)
    expect(wrapper).toMatchSnapshot()
  })
})
