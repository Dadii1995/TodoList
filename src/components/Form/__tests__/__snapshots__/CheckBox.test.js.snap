// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> render checkbox with initial value 1`] = `
<div
  className="form-check"
>
  <label
    className="form-check-label"
  >
    <input
      className="form-check-input"
      name="testCheckbox"
      onBlur={[Function]}
      onChange={[Function]}
      type="checkbox"
      value={true}
    />
     
  </label>
</div>
`;

exports[`<Checkbox/> render checkbox with label 1`] = `
<div
  className="form-check"
>
  <label
    className="form-check-label"
  >
    <input
      className="form-check-input"
      name="testCheckbox"
      onBlur={[Function]}
      onChange={[Function]}
      type="checkbox"
    />
     
    Accept terms
  </label>
</div>
`;

exports[`<Checkbox/> render checkbox without label 1`] = `
<div
  className="form-check"
>
  <label
    className="form-check-label"
  >
    <input
      className="form-check-input"
      name="testCheckbox"
      onBlur={[Function]}
      onChange={[Function]}
      type="checkbox"
    />
     
  </label>
</div>
`;
